### PR TITLE
Add templated data sources and dashboards for upload service

### PR DESCRIPTION
**Pull request by mweiden**

### CODE REVIEW

File: upload.tmp [hunk 1]
```diff
@@ -0,0 +1,1505 @@
...
+          "iconColor": "rgba(0, 211, 255, 1)",
+          "name": "Annotations & Alerts",
+          "type": "dashboard"
+        }
+      ]
+    },
+    "editable": true,
+    "gnetId": null,
+    "graphTooltip": 0,
+    "id": 9,
```
Comment (mweiden) [inline upload.tmp]: @parthshahva you have to delete the `"id"` key.

File: upload.tmp [hunk 2]
```diff
@@ -0,0 +1,1505 @@
...
+        "12h",
+        "24h",
+        "2d",
+        "7d",
+        "30d"
+      ]
+    },
+    "timezone": "",
+    "title": "Upload [Dev]",
+    "uid": "8a309-4ik",
```
Comment (mweiden) [inline upload.tmp]: ... and change this to something that is namespaced by `${var.env}`

File: terraform/modules/env-dashboards/upload-dashboard.tf [hunk 3]
```diff
@@ -0,0 +1,1486 @@
...
+          "alias": "Database Connections",
+          "dimensions": {
+            "DBInstanceIdentifier": "upload-cluster-dev-0"
+          },
+          "highResolution": false,
+          "metricName": "DatabaseConnections",
+          "namespace": "AWS/RDS",
+          "period": "600",
+          "refId": "B",
+          "region": "us-east-1",
```
Comment (mweiden) [inline terraform/modules/env-dashboards/upload-dashboard.tf]: I would template the values of the `"region"` keys as well

File: terraform/modules/env-dashboards/datasources.tf [hunk 4]
```diff
@@ -16,4 +17,40 @@ locals {
   "readOnly": false
 }
 EOF
+
+  aws_upload_datasource = <<EOF
+{
+  "name": "${local.aws_upload_datasource_name}",
+  "type": "postgres",
+  "url": "${data.external.upload_db_secrets_processing.result.username}",
```
Comment (MDunitz) [inline terraform/modules/env-dashboards/datasources.tf]: why are url, user and password all the same?

Comment (parthshahva) [reply terraform/modules/env-dashboards/datasources.tf]: lol good catch. I was testing something and pushed it up by accident. Thanks!

File: terraform/modules/env-dashboards/datasources.tf [hunk 5]
```diff
@@ -16,4 +17,40 @@ locals {
...
+data "aws_secretsmanager_secret" "upload_db_creds" {
+  name = "dcp/upload/${var.env}/database"
+}
+
+data "aws_secretsmanager_secret_version" "upload_db_creds" {
+  secret_id = "${data.aws_secretsmanager_secret.upload_db_creds.id}"
+}
+
+data "external" "upload_db_secrets_processing" {
+  program = ["sh", "${path.root}/../../../../terraform/modules/env-dashboards/process_upload_db_secrets.sh"]
```
Comment (mweiden) [inline terraform/modules/env-dashboards/datasources.tf:50]: I think the `${PROJECT_ROOT}/.bin` directory is linked into the container if you'd rather not snake this path into the modules dir...

**Pull request merged**